sdarot.py

```diff
@@ -74,40 +74,58 @@ def get_data(self, url, data=None):
         return None
 
     def get_video(self, video_url):
+        download_exist = False
+        file_size_offline = 0
+
+        # create directory for episode
+        episode_path = join(self.output_path,
+                            f'{self.serie_name}',
+                            f'Season-{self.season:02d}')
+        os.makedirs(episode_path, exist_ok=True)
+
+        # get the file name
+        filename = f'Episode-{self.episode:02d}.mp4'
+        offline_filename = join(episode_path, filename)
+
+        # if file exists set Range header to start stream from the file size
+        if os.path.isfile(offline_filename):
+            download_exist = True
+            file_size_offline = os.stat(offline_filename).st_size
+            print(f'Found existing episode: {filename}.')
+
+        self.s.headers.update({'Range': f'bytes={file_size_offline}-'})
 
         # get video with stream
         res = self.s.get(video_url, stream=True)
-        if not res or res.status_code != 200:
-            print(Fore.RED + 'Error occurred, no video')
+        if not res or res.status_code not in (200, 206, 416):
+            if res.status_code == 416:
+                print('Episode already downloaded - skipping download')
+                return True
```

Comment:
inner if is unreachable

Reply:
I know its a bit ugly but It is - check it out, just download something and run it again
It was a workaround to add it here to save code checking for specific errors before.

Reply:
I actually tried it and I get why it works